cmd/query/app/ui/assets_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package ui
+
+import (
+	"embed"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestGetStaticFiles_ReturnsPlaceholderWhenActualNotPresent(t *testing.T) {
+	// swap out the assets FS for an empty one and then replace it back when the
+	// test completes
+	currentActualAssets := actualAssetsFS
+	actualAssetsFS = embed.FS{}
+	t.Cleanup(func() { actualAssetsFS = currentActualAssets })
+
+	fs := GetStaticFiles(zap.NewNop())
```

Comment:
you could use pkg/testutils/logger to capture and verify log output. And you can make this test data-driven where you run it once with `embed.FS{}` (expecting warning log) and the 2nd time with placeholder fs (not expecting warning log)

Reply:
@yurishkuro i'm not sure I can use placeholder fs since it has `index.html` and not `index.html.gz`